### PR TITLE
Updated all links to sun.com reference to oracle site

### PR DESCRIPTION
**Pull request by damianszczepanik**
Issue #711

### CODE REVIEW

**Comment by romani** [comment]
Why did you choose java6 references ? Even java7 become eol in few months.

As java6 become unavailable, all that links will become brocken, and linkcheck maven plugin start complain

**Comment by damianszczepanik** [comment]
Totally agree. Upgraded all links to JDK8

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/SuperCloneCheck.java [hunk 1]
```diff
@@ -26,7 +26,7 @@
  * </p>
  * <p>
  * Reference:<a
- * href="http://java.sun.com/j2se/1.4.1/docs/api/java/lang/Object.html#clone()">
+ * href="http://www.oracle.com/technetwork/java/index.html#clone%28%29">
```
Comment (mkordas) [inline src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/SuperCloneCheck.java]: This link doesn't work - please recheck all.

**Comment by romani** [comment]
I think we need to try to enforce linkcheck plugin to catch such cases , might be helpful - http://maven.apache.org/plugins/maven-linkcheck-plugin/linkcheck-mojo.html#httpClientParameters

**Comment by romani** [comment]
@damianszczepanik, you can update pom.xml code like : 
```
      <plugin>
        <groupId>org.apache.maven.plugins</groupId>
        <artifactId>maven-linkcheck-plugin</artifactId>
        <version>1.2</version>
        <configuration>

<httpMethod>GET</httpMethod>
<timeout>3000</timeout>
<httpFollowRedirect>false</httpFollowRedirect>
....
```
to get list all problematic links, after you fix that, we could apply that config permanently to never have that problems in future.

**Comment by romani** [comment]
@damianszczepanik, are you going to provide a fix ? it is ok to drop issue, I just do not want to let issue hang in unclear state.

**Comment by damianszczepanik** [comment]
I will do this according to your suggestion. Sorry for delay.

**Comment by damianszczepanik** [comment]
It took me a while to validate whole checkstyle project but finally I've done it and [published](http://damianszczepanik.github.io/checkstyle/checkstyle/pull/724/linkcheck.xml).

I would not include this process to mvn install because each build would take ~20 minutes. You can consider to put it into .travis to validate over travis build file but this needs to be think over carefully so I won't add it now.

**Comment by romani** [comment]
@damianszczepanik, please remove extra config for linkcheck plugin, please update existing config.
I am reviewing your PR.....

**Comment by damianszczepanik** [comment]
@romani 
One of three build failed with message:

```
Failed to execute goal org.apache.maven.plugins:maven-site-plugin:3.4:site (default-site) on project checkstyle: Execution default-site of goal org.apache.maven.plugins:maven-site-plugin:3.4:site failed: Plugin org.apache.maven.plugins:maven-site-plugin:3.4 or one of its dependencies could not be resolved: Could not transfer artifact commons-collections:commons-collections:jar:3.2.1 from/to central (https://repo.maven.apache.org/maven2): GET request of: commons-collections/commons-collections/3.2.1/commons-collections-3.2.1.jar from central failed: Connection reset -> 
```
Can I re-trigger it somehow?

**Comment by romani** [comment]
merged as FF, THANKS A LOT !!!!

**Pull request closed without merging**